
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Calculate DNA methylation level &#8212; BatMeth2 1.1 documentation</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet" />
  <link href="../_static/css/index.c5995385ac14fb8791e8eb36b4908be2.css" rel="stylesheet" />

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" />
    
  <link rel="preload" as="script" href="../_static/js/index.1c5a1a01449ed65a7b51.js">

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/sphinx-book-theme.12a9622fbb08dcb3a2a40b2c02b83a57.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Calulate mC across predefined regions" href="MethyGff.html" />
    <link rel="prev" title="Alignment" href="Alignment.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
      
      
      <h1 class="site-logo" id="site-title">BatMeth2 1.1 documentation</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="install.html">
   Installation
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="example.html">
   Example Data
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="bdindex.html">
   Build index
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="Pipeline.html">
   Pipeline
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="Alignment.html">
   Alignment
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Calculate DNA methylation level
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="MethyGff.html">
   Calulate mC across predefined regions
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="bt2bigwig.html">
   Meth2BigWig
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="PlotMeth.html">
   PlotMeth
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="DiffMeth.html">
   DiffMeth
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="Requirements.html">
   Requirements
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../_sources/function/Calmeth.rst.txt"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.rst</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#calmeth">
   Calmeth
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#paramaters">
   Paramaters
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#output-files">
   Output files
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#output-file-format">
   Output file format
  </a>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="calculate-dna-methylation-level">
<h1>Calculate DNA methylation level<a class="headerlink" href="#calculate-dna-methylation-level" title="Permalink to this headline">¶</a></h1>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#calmeth" id="id1">Calmeth</a></p></li>
<li><p><a class="reference internal" href="#paramaters" id="id2">Paramaters</a></p></li>
<li><p><a class="reference internal" href="#output-files" id="id3">Output files</a></p></li>
<li><p><a class="reference internal" href="#output-file-format" id="id4">Output file format</a></p></li>
</ul>
</div>
<div class="section" id="calmeth">
<h2><a class="toc-backref" href="#id1">Calmeth</a><a class="headerlink" href="#calmeth" title="Permalink to this headline">¶</a></h2>
<p>Calculate DNA methylation level from alignment files, you can obtained single-base cytosine DNA
methylation results, and the chromosome region DNA methylation levels files.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>An example usage is:
  with bam file:
    calmeth <span class="o">[</span>options<span class="o">]</span> -g genome.fa  -b alignment.sort.bam -m output.methrario.txt
  with sam file:
    calmeth <span class="o">[</span>options<span class="o">]</span> -g genome.fa  -i alignment.sort.sam -m output.methrario.txt
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>The bam or sam file MUST sorted by <cite>samtools sort</cite>.</p>
</div>
</div>
<div class="section" id="paramaters">
<h2><a class="toc-backref" href="#id2">Paramaters</a><a class="headerlink" href="#paramaters" title="Permalink to this headline">¶</a></h2>
<table class="table">
<colgroup>
<col style="width: 4%" />
<col style="width: 17%" />
<col style="width: 79%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head" colspan="3"><p><strong>[ Main paramaters ]</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="2"><p>-m/–methratio</p></td>
<td><p>[MethFileNamePrefix]  Predix of methratio output file</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>–genome/-g</p></td>
<td><p>Name of the genome mapped against, MUST build index first <a class="reference internal" href="bdindex.html"><span class="doc">Build index</span></a></p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-i/–input</p></td>
<td><p>Sam format file, sorted by samtools sort.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-b/–binput</p></td>
<td><p>Bam format file, sorted by samtools sort.</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-Q [int]</p></td>
<td><p>caculate the methratio while read QulityScore &gt;= Q. default:20</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-n [float]</p></td>
<td><p>Number of mismatches, default 0.06 percentage of read length. [0-1]</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-c|–coverage</p></td>
<td><p>&gt;= &lt;INT&gt; coverage. default:4</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-nC</p></td>
<td><p>&gt;= &lt;INT&gt; Cs per region. default:1</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-R/–Regions</p></td>
<td><p>Bins for DMR caculate , default 1000(1kb) .</p></td>
</tr>
<tr class="row-odd"><td colspan="3"><p>–binsfile</p></td>
</tr>
<tr class="row-even"><td></td>
<td colspan="2"><p>DNA methylation level distributions in chrosome, default output file: {Prefix}.methBins.txt</p></td>
</tr>
<tr class="row-odd"><td colspan="3"><p>-s/–step</p></td>
</tr>
<tr class="row-even"><td></td>
<td colspan="2"><p>Chrosome using an overlapping sliding window of 100000bp at a step of 50000bp.
default step: 50000(bp)</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-r/–remove_dup</p></td>
<td><p>REMOVE_DUP, default:true</p></td>
</tr>
<tr class="row-even"><td colspan="3"><p>-f|–sam [outfile]</p></td>
</tr>
<tr class="row-odd"><td></td>
<td colspan="2"><p>f for sam format outfile contain methState.</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>–sam-seq-beforeBS</p></td>
<td><p>Converting BS read to the genome sequences.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>–help/-h</p></td>
<td><p>Print help</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="output-files">
<h2><a class="toc-backref" href="#id3">Output files</a><a class="headerlink" href="#output-files" title="Permalink to this headline">¶</a></h2>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="m">1</span>. prefix.methratio.txt
<span class="m">2</span>. prefix.methBins.txt
<span class="m">3</span>. prefix_Region.CG/CHG/CHH.txt
<span class="m">4</span>. prefix.mCdensity.txt
<span class="m">5</span>. prefix.mCcatero.txt
</pre></div>
</div>
</div>
<div class="section" id="output-file-format">
<h2><a class="toc-backref" href="#id4">Output file format</a><a class="headerlink" href="#output-file-format" title="Permalink to this headline">¶</a></h2>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="m">1</span>. methratio
    Chromosome Loci Strand Context C_count CT_count methlevel eff_CT_count rev_G_count rev_GA_count MethContext 5context
    <span class="c1"># ex. Chr1    61      +       CHH     3       11      0.286364        10.5    20      21      hU      ATCTT</span>
    <span class="c1"># C_count      The number of C in this base pair.</span>
    <span class="c1"># CT_count     The number of coverage in this base pair.</span>
    <span class="c1"># eff_CT_count Adjust read coverage based on opposite strand.</span>
    <span class="c1"># rev_G_count  The number of G in the reverse strand.</span>
    <span class="c1"># rev_GA_count The number of coverage in the reverse strand.</span>
    <span class="c1"># MethContext  M/Mh/H/hU/U, M means the methylation level ≥ 80%, etc</span>
<span class="m">2</span>. methBins
    Chrom BinIndex methlevel context
    <span class="c1"># ex. Chr1    1       0.113674        CG</span>
    <span class="c1"># The BinIndex is defined by -s paramater in calmeth.</span>
    <span class="c1"># This file can be used for visualization the DNA methylation level acorss the chromosome.</span>
<span class="m">3</span>. Region
    chrom regionStart strand context c_count ct_count
    <span class="c1"># ex. Chr1    1001    +       CG      1       227</span>
    <span class="c1"># The bins methylation level output file (BS.mr_Region.C*.txt) can be used to do DMR detection.</span>
<span class="m">4</span>. mCdensity
    CG/CHG/CHH C count <span class="k">in</span> <span class="o">[</span><span class="m">0</span>, <span class="m">1</span>%<span class="o">)</span> <span class="o">[</span><span class="m">1</span>%, <span class="m">2</span>%<span class="o">)</span> ... <span class="o">[</span><span class="m">49</span>%, <span class="m">50</span>%<span class="o">)</span> ... <span class="o">[</span><span class="m">99</span>%, <span class="m">100</span>%<span class="o">]</span>
    <span class="c1"># According to the DNA methylation level, the number of cytosine sites at different methylation levels was counted from 0 to 100.</span>
<span class="m">5</span>. mCcatero
    Average DNA methylation level including mC, mCG and other states.
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>For feature requests or bug reports please open an issue <a class="reference external" href="http://github.com/ZhouQiangwei/BatMeth2">on github</a>.</p>
</div>
</div>
</div>


              </div>
              
        
            



<div class='prev-next-bottom'>
    
    <div id="prev">
        <a class="left-prev" href="Alignment.html" title="previous page">
            <i class="prevnext-label fas fa-angle-left"></i>
            <div class="prevnext-info">
                <p class="prevnext-label">previous</p>
                <p class="prevnext-title">Alignment</p>
            </div>
        </a>
    </div>
     <div id="next">
        <a class="right-next" href="MethyGff.html" title="next page">
            <div class="prevnext-info">
                <p class="prevnext-label">next</p>
                <p class="prevnext-title">Calulate mC across predefined regions</p>
            </div>
            <i class="prevnext-label fas fa-angle-right"></i>
        </a>
     </div>

</div>
        
        </div>
    </div>
    <footer class="footer">
    <div class="container">
      <p>
        
          By momocoding<br/>
        
            &copy; Copyright 2020, momocoding.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.1c5a1a01449ed65a7b51.js"></script>

  
  </body>
</html>