
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Calulate mC across predefined regions &#8212; BatMeth2 1.1 documentation</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet" />
  <link href="../_static/css/index.c5995385ac14fb8791e8eb36b4908be2.css" rel="stylesheet" />

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" />
    
  <link rel="preload" as="script" href="../_static/js/index.1c5a1a01449ed65a7b51.js">

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/sphinx-book-theme.12a9622fbb08dcb3a2a40b2c02b83a57.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Meth2BigWig" href="bt2bigwig.html" />
    <link rel="prev" title="Calculate DNA methylation level" href="Calmeth.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
      
      
      <h1 class="site-logo" id="site-title">BatMeth2 1.1 documentation</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="install.html">
   Installation
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="example.html">
   Example Data
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="bdindex.html">
   Build index
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="Pipeline.html">
   Pipeline
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="Alignment.html">
   Alignment
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="Calmeth.html">
   Calculate DNA methylation level
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Calulate mC across predefined regions
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="bt2bigwig.html">
   Meth2BigWig
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="PlotMeth.html">
   PlotMeth
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="DiffMeth.html">
   DiffMeth
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="Requirements.html">
   Requirements
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../_sources/function/MethyGff.rst.txt"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.rst</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#methygff">
   methyGff
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#paramaters">
   Paramaters
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#output-files">
   Output files
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#output-format">
   Output format
  </a>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="calulate-mc-across-predefined-regions">
<h1>Calulate mC across predefined regions<a class="headerlink" href="#calulate-mc-across-predefined-regions" title="Permalink to this headline">¶</a></h1>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#methygff" id="id1">methyGff</a></p></li>
<li><p><a class="reference internal" href="#paramaters" id="id2">Paramaters</a></p></li>
<li><p><a class="reference internal" href="#output-files" id="id3">Output files</a></p></li>
<li><p><a class="reference internal" href="#output-format" id="id4">Output format</a></p></li>
</ul>
</div>
<div class="section" id="methygff">
<h2><a class="toc-backref" href="#id1">methyGff</a><a class="headerlink" href="#methygff" title="Permalink to this headline">¶</a></h2>
<p>Through GTF, GFF or bed files, the methylation level of the designated region and
upstream and downstream was calculated, and the methylation level matrix was generated.
The generated methylation level file and matrix file can be used to generate profile and Heatmap
visualization.</p>
<ul class="simple">
<li><p>The methratio file calculated by <span class="xref std std-doc">calmeth</span>, the format is <cite>chrom pos strand context nC nCover methlevel</cite>.</p></li>
</ul>
<p>For exsample: <cite>chr1    34      -       CHG     2       14      0.142857</cite></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>An example usage is:
  with gtf file:
    methyGff -B -o gene.meth -G genome.fa -gtf gene.gtf -m output.methrario.txt

  with multiple gtf file:
    methyGff -B -o expressed.gene.meth unexpressed.gene.meth <span class="se">\</span>
        -G genome.fa  -gtf expressed.gene.gtf unexpressed.gene.gtf -m output.methrario.txt

  with bed file:
    methyGff -B -o gene.meth -G genome.fa -b gene.bed -m output.methrario.txt

  with multiple bed file:
    methyGff -B -o expressed.gene.meth unexpressed.gene.meth <span class="se">\</span>
        -G genome.fa -b expressed.gene.bed unexpressed.gene.bed -m output.methrario.txt
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>The number of input gtf/gff/bed files must be the same as the number of output prefixes..</p>
</div>
</div>
<div class="section" id="paramaters">
<h2><a class="toc-backref" href="#id2">Paramaters</a><a class="headerlink" href="#paramaters" title="Permalink to this headline">¶</a></h2>
<p>Command Format :   methyGff [options] -o &lt;OUT_PREFIX&gt; -G GENOME -gff &lt;GFF file&gt;/-gtf &lt;GTF file&gt;/-b &lt;bed file&gt; -m &lt;from Split methratio outfile&gt; [-B][-P]</p>
<table class="table">
<colgroup>
<col style="width: 4%" />
<col style="width: 17%" />
<col style="width: 79%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head" colspan="3"><p><strong>[ Main paramaters ]</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="2"><p>-o/–out</p></td>
<td><p>Output file prefix</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>–genome/-G</p></td>
<td><p>Name of the genome mapped against, MUST build index first <a class="reference internal" href="bdindex.html"><span class="doc">Build index</span></a></p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-m|–methratio</p></td>
<td><p>DNA methratio output file, generated by the tool <span class="xref std std-doc">calmeth</span></p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-c|–coverage</p></td>
<td><p>&gt;= &lt;INT&gt; coverage. default:4</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-C</p></td>
<td><p>&lt;= &lt;INT&gt; coverage. default 600.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-nC</p></td>
<td><p>&gt;= &lt;INT&gt; Cs per bins or genes. default:1</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-gtf/-gff</p></td>
<td><p>Gtf/gff file</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-b</p></td>
<td><p>Bed file, chrom start end</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-b4</p></td>
<td><p>Bed file, chrom start end strand</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-b5</p></td>
<td><p>Bed file, chrom start end geneid strand</p></td>
</tr>
<tr class="row-even"><td colspan="3"><p>-d/–distance</p></td>
</tr>
<tr class="row-odd"><td></td>
<td colspan="2"><p>DNA methylation level distributions in body and &lt;INT&gt;-bp flanking sequences.
The distance of upstream and downstream. default:2000</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-B/–body</p></td>
<td><p>Calculate the DNA methylation level of per region.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-P/–promoter</p></td>
<td><p>Calculate the DNA methylation level of per region’s upstream [d]k.</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>–TSS</p></td>
<td><p>Caculate matrix for TSS. [Outfile: outPrefix.TSS.cg.txt]</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>–TTS</p></td>
<td><p>Caculate matrix for TTS. [Outfile: outPrefix.TTS.cg.n.txt]</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>–GENE</p></td>
<td><p>Caculate matrix for TSS. [Outfile: outPrefix.TSS.cg.txt]</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>–TTS</p></td>
<td><p>Caculate matrix for GENE and flank [d]k. [outPrefix.GENE.cg.txt]</p></td>
</tr>
<tr class="row-even"><td colspan="3"><p>-s/–step</p></td>
</tr>
<tr class="row-odd"><td></td>
<td colspan="2"><p>Gene body and their flanking sequences using an overlapping sliding window of 2% of
the sequence length at a step of 1% of the sequence length. So default step: 0.01 (1%)</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-bl/–bodyLen</p></td>
<td><p>Body length to which all regions will be fit. (default: same as -d)</p></td>
</tr>
<tr class="row-odd"><td colspan="3"><p>-S/–chromStep</p></td>
</tr>
<tr class="row-even"><td></td>
<td colspan="2"><p>Caculate the density of genes/TEs in chromsome using an overlapping sliding window of
100000bp at a step of 50000bp, must equal “-s” in Split.. default step: 50000(bp)</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>–help/-h</p></td>
<td><p>Print help</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="output-files">
<h2><a class="toc-backref" href="#id3">Output files</a><a class="headerlink" href="#output-files" title="Permalink to this headline">¶</a></h2>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>Output</p>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="m">1</span>. prefix.meth.AverMethylevel.txt
<span class="m">2</span>. prefix.meth.Methylevel.txt
<span class="m">3</span>. prefix.meth.TSSprofile.txt
<span class="m">4</span>. prefix.meth.centerprofile.txt
<span class="m">5</span>. prefix.col-0.meth.annoDensity.txt
<span class="m">6</span>. prefix.meth.body.c*.txt
<span class="c1"># run methyGff with -B paramater</span>
<span class="m">7</span>. prefix.bdgene.Promoter.c*.txt
<span class="c1"># run methyGff with -P paramater</span>
<span class="m">8</span>. prefix.bdgene.TSS.cg.txt
<span class="c1"># run methyGff with --TSS paramater</span>
<span class="m">9</span>. prefix.bdgene.TTS.cg.txt
<span class="c1"># run methyGff with --TTS paramater</span>
<span class="m">10</span>. prefix.bdgene.GENE.cg.txt
<span class="c1"># run methyGff with --GENE paramater</span>
</pre></div>
</div>
</div>
<div class="section" id="output-format">
<h2><a class="toc-backref" href="#id4">Output format</a><a class="headerlink" href="#output-format" title="Permalink to this headline">¶</a></h2>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="m">1</span>. AverMethylevel
    CG/CHG/CHH meth_of_bin1 bin2 bin3 ... bini binj ... binN
    <span class="c1">## N is defined by -s paramater</span>
<span class="m">2</span>. Methylevel
    CG/CHG/CHH UP/BODY/DOWN Methylevel
    <span class="c1"># per line means 1 gene/TE/region</span>
<span class="m">3</span>. TSSprofile
    CG/CHG/CHH meth_of_bin1 bin2 bin3 ... bini binj ... binN
    <span class="c1"># DNA methylation level across TSS</span>
    <span class="c1"># -d N kb, TSS upstream and downstream N kb</span>
    <span class="c1"># -s move step</span>
<span class="m">4</span>. centerprofile
    CG/CHG/CHH meth_of_bin1 bin2 bin3 ... bini binj ... binN
    <span class="c1"># DNA methylation level across region center</span>
    <span class="c1"># -d N kb, center point upstream and downstream N kb</span>
    <span class="c1"># -s move step</span>
<span class="m">5</span>. annoDensity
    chrom pos methlevel strand
    <span class="c1"># ex. Chr1    0       0.559940        +-</span>
    <span class="c1"># The density of region distributions on chromsome</span>
<span class="m">6</span>. body/Promoter
    chrom regionStart strand context C_count CT_count regionID
    <span class="c1"># ex. Chr1    3631    +       CG      45      1314    AT1G01010</span>
    <span class="c1"># This file can be used for visualization using `PlotMeth:bt2profile` or `PlotMeth:bt2heatmap`</span>
<span class="m">7</span>. TSS/TTS/GENE
    regionID meth_of_bin1 bin2 bin3 ... bini binj ... binN
    ... ...
    <span class="c1"># This file is the methylation matrix across all genes, per line represents one region (gene/TE/etc)</span>
    <span class="c1"># This file can be used for visualization using `PlotMeth:bt2heatmap`</span>
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>For feature requests or bug reports please open an issue <a class="reference external" href="http://github.com/ZhouQiangwei/BatMeth2">on github</a>.</p>
</div>
</div>
</div>


              </div>
              
        
            



<div class='prev-next-bottom'>
    
    <div id="prev">
        <a class="left-prev" href="Calmeth.html" title="previous page">
            <i class="prevnext-label fas fa-angle-left"></i>
            <div class="prevnext-info">
                <p class="prevnext-label">previous</p>
                <p class="prevnext-title">Calculate DNA methylation level</p>
            </div>
        </a>
    </div>
     <div id="next">
        <a class="right-next" href="bt2bigwig.html" title="next page">
            <div class="prevnext-info">
                <p class="prevnext-label">next</p>
                <p class="prevnext-title">Meth2BigWig</p>
            </div>
            <i class="prevnext-label fas fa-angle-right"></i>
        </a>
     </div>

</div>
        
        </div>
    </div>
    <footer class="footer">
    <div class="container">
      <p>
        
          By momocoding<br/>
        
            &copy; Copyright 2020, momocoding.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.1c5a1a01449ed65a7b51.js"></script>

  
  </body>
</html>